### PR TITLE
:hammer: Add `mypy` Checks

### PR DESCRIPTION
**Pull request by Jiaqi-Lv**
Files added:
- `tiatoolbox/utils/exceptions.py`
- `tiatoolbox/utils/magic.py`
- `tiatoolbox/utils/metrics.py` 
- `tiatoolbox/utils/transforms.py`
- `tiatoolbox/tiatoolbox.py`

### CODE REVIEW

File: tiatoolbox/utils/transforms.py [hunk 1]
```diff
@@ -258,9 +262,13 @@ def bounds2locsize(
     left, top, right, bottom = bounds
     origin = origin.lower()
     if origin == "upper":
-        return np.array([left, top]), np.array([right - left, bottom - top])
+        return np.ndarray(
```
Comment (shaneahmed) [inline tiatoolbox/utils/transforms.py:265]: I do not understand why you are doing this.

Comment (shaneahmed) [inline tiatoolbox/utils/transforms.py:265]:
```suggestion
        return np.array(
```

File: tiatoolbox/utils/transforms.py [hunk 2]
```diff
@@ -258,9 +262,13 @@ def bounds2locsize(
...
+            [np.array([left, top]), np.array([right - left, bottom - top])],
+        )
     if origin == "lower":
-        return np.array([left, bottom]), np.array([right - left, top - bottom])
+        return np.ndarray(
```
Comment (shaneahmed) [inline tiatoolbox/utils/transforms.py:269]:
```suggestion
        return np.array(
```

**Comment by shaneahmed** [comment]
@Jiaqi-Lv Please can you fix the coverage?
https://app.codecov.io/gh/TissueImageAnalytics/tiatoolbox/pull/740/blob/tiatoolbox/utils/transforms.py#L110

File: pyproject.toml [hunk 3]
```diff
@@ -174,4 +174,3 @@ max-args = 10
 
 [tool.mypy]
 ignore_missing_imports = true
-python_version = 3.8
```
Comment (shaneahmed) [inline pyproject.toml:177]: Why this has been removed?

File: tiatoolbox/typing.py [hunk 4]
```diff
@@ -6,7 +6,8 @@
 import numpy as np
 from shapely.geometry import LineString, Point, Polygon  # type: ignore[import-untyped]
 
-# Proper type annotations for shapely is not yet available
+# Proper type annotations for shapely is not yet avaliable
```
Comment (shaneahmed) [inline tiatoolbox/typing.py:9]:
```suggestion
# Proper type annotations for shapely is not yet avaliable.
```

File: pyproject.toml (hunk 3 above)
Comment (Jiaqi-Lv) [reply pyproject.toml:177]: Removed it for testing, forgot to add it back. It's back now

**Review by shaneahmed — APPROVED** [review]
Thanks @Jiaqi-Lv

**Pull request merged**